backend/routes/partners.py

```diff
@@ -193,3 +203,278 @@ def add_member_to_partner(partner_id: int):
         "role": partner_member.role,
     })
     return partner_member_orm_to_json(created)
+
+
+# user can join org they were invited to
+@bp.route("/join", methods=["POST"])
+@jwt_required()
+@min_role_required(UserRole.PUBLIC)
+def join_organization():
+    try:
+        body = request.get_json()
+        user_exists = PartnerMember.query.filter_by(
+            user_id=body["user_id"],
+            partner_id=body["partner_id"]).first()
+        if user_exists:
+            return {
+                "status" : "Error",
+                "message": "User already in the organization"
+            }, 400
+        else:
+            new_member = PartnerMember(
+                user_id=body["user_id"],
+                partner_id=body["partner_id"],
+                role=body["role"],
+                date_joined=datetime.now(),
+                is_active=True
+            )
+            db.session.add(new_member)
+            db.session.commit()
+            Invitation.query.filter_by(
+                user_id=body["user_id"],
+                partner_id=body["partner_id"]).delete()
+            db.session.commit()
+            return {
+                "status": "ok",
+                "message": "Successfully joined partner organization"
+            } , 200
+    except Exception:
+        db.session.rollback()
+        return {
+            "status": "Error",
+            "message": "Something went wrong!"
+        }, 500
+    finally:
+        db.session.close()
+
+
+# user can leave org they already joined
+@bp.route("/leave", methods=["DELETE"])
+@jwt_required()
+@min_role_required(UserRole.PUBLIC)
+def leave_organization():
+    """
+    remove from PartnerMember table
+    """
+    try:
+        body = request.get_json()
+        result = PartnerMember.query.filter_by(
+            user_id=body["user_id"], partner_id=body["partner_id"]).delete()
+        db.session.commit()
+        if result > 0:
+            return {
+                "status": "ok",
+                "message": "Succesfully left organization"
+            }, 200
+        else:
+            return {
+                "status": "Error",
+                "message": "Not a member of this organization"
+            }, 400
+    except Exception:
+        db.session.rollback()
+        return {
+            "status": "Error",
+            "message": "Something went wrong!"
+        }
+    finally:
+        db.session.close()
+
+
+# inviting anyone to NPDC
```

Comment:
We should probably just fold this into the existing "Add member" endpoint. Otherwise, we kind of have an end-run around the invitation process.

Reply:
Do you mean shifting the code for the current "invite_user" function/endpoint to  "add_member_to_partner" function/endpoint and deleting the "invite_user" one? Since both of those have the same purpose.

Reply:
Yes, exactly